### ISSUE TITLE
create tests for simple_transformer_text_classification taskmodule from pytorch-ie

### ISSUE THREAD

**Issue by ArneBinder** [ArneBinder/pie-document-level#29]
This requires to write individual tests for all functions in that taskmodule (@akashkumar25 please finish the list):
 - [x] the taskmodule creation
 - [x]  `_prepare`
 - [x] `_post_prepare`
 - [x] `encode_input`
 - [x] `encode_target`
 - [x] `collate`
 - [x] `unbatch_output`
 - [x] `create_annotations_from_output`

Each test should be combined with a respective fixture function that creates the actual output. Then, the test function gets the fixture as input and checks for the correct values. E.g.:
```python
@pytest.fixture(scope="module")
def taskmodule():
  return SimpleTransformerTextClassificationTaskModule(tokenizer_name_or_path="bert-base-uncased")

def test_taskmodule(taskmodule):
  assert taskmodule is not None
  # anything else to assert, if necessary

# similar for fixture "taskmodule_prepared" and its test "test_taskmodule_prepared" etc. 
```

Note: Take inspiration from the [tests for the seq2seq taskmodule](https://github.com/ChristophAlt/pytorch-ie/blob/main/tests/taskmodules/test_transformer_seq2seq.py).

**Comment by akashkumar25** [comment]
### Documentation

## Fixtures

 `config`

- This fixture is defined with the @pytest.fixture decorator and is used to provide a configuration dictionary (config) to other tests.
- It uses the CONFIGS_DICT dictionary to retrieve the configuration based on the parameter passed to the fixture (request.param).

`config_str `

- This fixture takes the config fixture as a parameter and returns a string representation of the configuration dictionary (config).
- It uses the _config_to_str function to convert the dictionary into a string.

`pre_prepared_taskmodule`

- This fixture creates an instance of the SimpleTransformerTextClassificationTaskModule class with the specified tokenizer and configuration.
- It takes the config fixture as a parameter to retrieve the configuration values.


`documents`

- This fixture creates a list of ExampleDocument objects with predefined texts.
- It also assigns labels to the documents using the Label class and appends them to the respective documents.
- The fixture returns the list of created documents.

`taskmodule`

- This fixture takes the pre_prepared_taskmodule fixture and the documents fixture as parameters.
- It prepares the task module for the given documents by calling the _prepare and _post_prepare methods of the pre_prepared_taskmodule.
- The prepared task module is then returned by the fixture.

`Other fixtures`

- The remaining fixtures (task_encoding_without_targets, target, task_encoding, batch, unbatched_outputs, annotations_from_output) provide setup data or objects required for specific tests.
- They take the taskmodule and other relevant objects as parameters and perform necessary operations to prepare the data or objects needed for the corresponding tests.

## Tests

`test_taskmodule`

- Verifies that the pre_prepared_taskmodule object is not None.
- Ensures that the pre_prepared_taskmodule is successfully created.

`test_post_prepare`

- Verifies that the taskmodule object is not None.
- Asserts that the taskmodule is prepared.
- Checks the correctness of the label_to_id mapping.
- Checks the correctness of the id_to_label mapping.

`test_encode_input`

- Encodes the input for the first document in the documents list using the taskmodule.
- Checks that the encoded input is not None.
- Asserts that the encoded input corresponds to the first document.
- Ensures that the encoded input does not have any targets.
- Verifies that the encoded input contains the expected input fields.
- Validates that the metadata of the encoded input is empty.
- Checks the correctness of the input tokens, attention mask, and token type IDs.

`test_target`

- Encodes the target label for the task_encoding_without_targets using the taskmodule.
- Checks the correctness of the encoded target label.

test_task_encoding

- Verifies that the task_encoding object is not None.

`test_collate`

- Collates a batch of task_encoding_without_targets using the taskmodule.
- Checks the correctness of the collated batch encoding, including input IDs, attention mask, and token type IDs.

`test_unpatch_output`

- Verifies that the unbatched_outputs object is not None.
- Checks the correctness of the unbatched model output.

`test_annotations_from_output`

- Creates named annotations from the task_encoding_without_targets and unbatched_outputs using the taskmodule.
- Checks the correctness of the created named annotations.

**Comment by ArneBinder** [comment]
The was integrated upstream, see https://github.com/ChristophAlt/pytorch-ie/pull/282.
